lib/models/addon.js

```diff
@@ -1178,7 +1173,28 @@ let addonProto = {
     @return {Tree} Preprocessed javascript
   */
   preprocessJs() {
-    return preprocessJs.apply(preprocessJs, arguments);
+    let emberCLIBabelConfigKey = this._emberCLIBabelConfigKey();
+
+    let original = this.options[emberCLIBabelConfigKey];
+
+    // the app will handle transpilation after it tree-shakes
+    // do it here instead of the constructor because
+    // ember-data and others do their own compilation in their
+    // treeForAddon without calling super
+    // they need the original params preserved because they call
+    // babel themselves and expect compilation the old way
+    this.options[emberCLIBabelConfigKey] = Object.assign({}, original, {
```

Comment:
This really should be moved back to the constructor if at all possible, setting it here means that others that depend on it earlier just don't have access to the values when they expect...

Reply:
I remember why I placed it here instead of the constructor. Having it in the constructor means I need to handle the legacy /modules using broccoli-module-normalizer. Making that change now.

Reply:
done